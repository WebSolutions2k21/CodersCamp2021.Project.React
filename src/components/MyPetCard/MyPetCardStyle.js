import { makeStyles } from '@mui/styles';

export const useStyles = makeStyles({
  root: {
    minWidth: 250
  },

  title: {
    textAlign: 'center',
    textTransform: 'uppercase',
  },

  ico: {
    width: '1.25rem',
    height: '1.25rem',
    padding: '0 10px',
  },

  pos: {
    color: '#112025',
    textTransform: 'uppercase',
  },

  small: {
    textTransform: 'lowercase',
  },
});
