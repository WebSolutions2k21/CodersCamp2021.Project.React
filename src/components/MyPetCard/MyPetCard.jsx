import { Typography, CardContent, Card, Grid } from '@mui/material';
import { useStyles } from './MyPetCardStyle';
import { ReactComponent as Dog } from '../../assets/dog.svg';

export const MyPetCard = ({ name, type, breed, age }) => {
  const classes = useStyles();

  return (
    <Card data-testid="my-pet-form">
      <CardContent>
        <Typography className={classes.title} alignItems="center" color="#16bac6" variant="h5">
          <Dog className={classes.ico} />
          {name}
        </Typography>
        <Grid>
          <Typography className={classes.pos} display="inline">
            Type:{' '}
          </Typography>
          <Typography color="textSecondary" className={classes.small} display="inline">
            {type}
          </Typography>
        </Grid>
        <Grid>
          <Typography className={classes.pos} display="inline">
            Breed:{' '}
          </Typography>
          <Typography display="inline" color="textSecondary" className={classes.small}>
            {breed}
          </Typography>
        </Grid>
        <Grid>
          <Typography className={classes.pos} display="inline">
            Birth Data:{' '}
          </Typography>
          <Typography color="textSecondary" className={classes.small} display="inline">
            {age.toString()}
          </Typography>
        </Grid>
      </CardContent>
    </Card>
  );
};
