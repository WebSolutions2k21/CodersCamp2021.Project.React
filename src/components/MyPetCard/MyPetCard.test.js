import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { MyPetCard } from './MyPetCard';


afterEach(() => {
  cleanup();
});

test('1. should render MyPetForm component', () => {
  render(<MyPetCard name="kapsel" type="dog" bread="mops" age="2" />);
  const inputComponent = screen.getByTestId("my-pet-form");
  expect(inputComponent).toBeInTheDocument();
  expect(inputComponent).not.toBeNull();
  expect(inputComponent).not.toBeDisabled();
  expect(inputComponent).toHaveTextContent('dog');
});

test('2. should render MyPetForm component', () => {
  render(<MyPetCard name="bu" type="cat" bread="sfinks" age="8" />);
  const inputComponent = screen.getByTestId("my-pet-form");
  expect(inputComponent).toBeInTheDocument();
  expect(inputComponent).not.toBeNull();
  expect(inputComponent).not.toBeDisabled();
  expect(inputComponent).toHaveTextContent('cat');
});

test('matches snapshot', () => {
  const tree = renderer.create(<MyPetCard name="kapsel" type="dog" bread="mops" age="2" />).toJSON();
  expect(tree).toMatchSnapshot();
});