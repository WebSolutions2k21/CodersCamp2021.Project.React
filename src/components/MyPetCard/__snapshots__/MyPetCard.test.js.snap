// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-testid="my-pet-form"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 makeStyles-title-12 css-rjgee4-MuiTypography-root"
    >
      <svg
        className="makeStyles-ico-13"
      >
        dog.svg
      </svg>
      kapsel
    </h5>
    <div
      className="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 makeStyles-pos-14 css-1sx1eu0-MuiTypography-root"
      >
        Type:
         
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 makeStyles-small-15 css-1goz70x-MuiTypography-root"
      >
        dog
      </p>
    </div>
    <div
      className="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 makeStyles-pos-14 css-1sx1eu0-MuiTypography-root"
      >
        Breed:
         
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 makeStyles-small-15 css-1mwruq2-MuiTypography-root"
      />
    </div>
    <div
      className="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 makeStyles-pos-14 css-1sx1eu0-MuiTypography-root"
      >
        Birth Data:
         
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 makeStyles-small-15 css-1goz70x-MuiTypography-root"
      >
        2
      </p>
    </div>
  </div>
</div>
`;
